File: Editorial/380_Insert_Delete_GetRandom_O_1/Python/1_HashMap_ArrayList.py
def insert(self, val: int) -> bool:
    """
    Inserts a value to the set. Returns true if the set did not already contain the specified element.
    """
    if val in self.dict:
        return False
    self.dict[val] = len(self.list)
    self.list.append(val)
    return True

File: Editorial/380_Insert_Delete_GetRandom_O_1/Java/1_HashMap_ArrayList.java
/** Inserts a value to the set. Returns true if the set did not already contain the specified element. */
public boolean insert(int val) {
  if (dict.containsKey(val)) return false;
    
  dict.put(val, list.size());
  list.add(list.size(), val);
  return true;
}
